pkg/controller/controller.go

```diff
@@ -341,3 +342,17 @@ func getEmail(r *http.Request) (string, error) {
 
 	return split[1], nil
 }
+
+// GetCatagoriesDisplayData takes validators as input and returns UIData for display. If there's an error during the GetUIData RPC call, it will be thrown.
+func getCatagoriesDisplayData(validators map[string]jvspb.Validator) (map[string]*jvspb.UIData, error) {
+	displayData := make(map[string]*jvspb.UIData)
+
+	for k, v := range validators {
+		d, err := v.GetUIData(context.Background(), &jvspb.GetUIDataRequest{})
```

Comment:
context should be passed in from [here](https://github.com/abcxyz/jvs/blob/3e1a165416e738a0b8ab6da57577b21ef8d36b2e/pkg/ui/server.go#L52). Actually I am not sure whether rpc call can be made at this stage, should it wait till the server and plugin are started successfully? Another option is to populate it at the first call. @yolocs what do you think?

Reply:
Your initial point is correct - this function should take the `ctx` from `New` and thread it through here.

Reply:
Thank you for your suggestion! 

Updated in the latest version to take ctx from New method from [here](https://github.com/abcxyz/jvs/blob/3e1a165416e738a0b8ab6da57577b21ef8d36b2e/pkg/ui/server.go#L52)